### ISSUE TITLE
ImageNet SOTA (BASIC-L+Lion finetune)

### ISSUE THREAD

**Issue by zc12345** [chaos-moon/paper_daily#14]
### ImageNet SOTA

inspiration: 

* 学习一下最新的[ImageNet sota](https://paperswithcode.com/sota/image-classification-on-imagenet)用了哪些技术

#### BASIC

* paper: [Combined Scaling for Zero-shot Transfer Learning](https://arxiv.org/abs/2111.10050)
* method
    * 方法本身是类似CLIP和ALIGN的image-text对比学习
    * contribution
        * 更多数据：6.6B脏数据image-text pair(=4xALIGN/16xCLIP)
        * 更大模型：3B参数(=3.75x参数，8xFLOPS)
        * 更大batch siz：65535(=4xALIGN/2xCLIP)
        * insight：在image-text对比学习中更大batch size带来更好性能

#### Model soups

* paper: [Model soups: averaging weights of multiple fine-tuned models improves accuracy without increasing inference time](https://arxiv.org/abs/2203.05482v3)
* code: [model-soups](https://github.com/mlfoundations/model-soups)
* method
    * 对不同超参下训出来的model
        * uniform soup: 将模型超参累加之后求平均
            * ![image](https://user-images.githubusercontent.com/15855347/234288453-b2f7bec2-fc2b-4174-8a77-27388cdac6f4.png)
        * greedy soup: 从best模型开始，每次滑动平均加入新模型参数，如果acc变好就更新参数
            * ![image](https://user-images.githubusercontent.com/15855347/234288401-2794d360-3772-4a1b-b6be-678543038a8f.png)

#### Lion Optimizer

* paper: [Symbolic Discovery of Optimization Algorithms](https://arxiv.org/abs/2302.06675)
* code: 
    * [timm.optim.lion](https://github.com/huggingface/pytorch-image-models/blob/main/timm/optim/lion.py) 
    * [lion](https://github.com/google/automl/tree/master/lion)
* method
    * 相较于之前的AdamW需要存储一阶和二阶动量，Lion只需要存一阶动量
    * ![image](https://user-images.githubusercontent.com/15855347/234288316-209f61b1-d844-4bfb-8571-7012c5358a29.png)
    * pseudo code
```python
def train(weight, gradient, momentum, lr): 
    update = interp(gradient, momentum, β1) 
    update = sign(update) 
    momentum = interp(gradient, momentum, β2) 
    weight_decay = weight * λ 
    update = update + weight_decay 
    update = update * lr 
    return update, momentum
```
* 使用Lion optimizer进行finetune之后的BASIC-L模型，相较于之前使用Adafactor优化器的ImageNet 90.9%->91.1%
